include/util.h

```diff
@@ -135,10 +137,27 @@ int PURE strncmp(const char *s1, const char *s2, int n);
 long CONST char_to_long(char c);
 long PURE str_to_long(const char *str);
 
-// Library functions for counting leading/trailing zeros.
-// GCC's builtins will emit calls to these functions when the platform
-// does not provide suitable inline assembly.
-// We only emit function definitions if CONFIG_CLZL_IMPL etc are set.
+/* Library functions for counting leading/trailing zeros.
+ * GCC provides the following builtin function for this:
+ *   __builtin_clz(unsigned int x)
+ *   __builtin_clzl(unsigned long x)
+ * At https://gcc.gnu.org/onlinedocs/gccint/Integer-library-routines.html it
+ * says that GCC will call helper functions in case the hardware does not
+ * provide instruction for counting leading/trailing zeros:
+ *   int __clzsi2 (unsigned int a)
+ *   int __clzdi2 (unsigned long a)
+ *   int __ctzsi2 (unsigned int a)
+ *   int __ctzdi2 (unsigned long a)
```

Comment:
The only reason we should be mentioning the GCC docs page is to point out that it's misleading, i.e. that the types it shows for these operations are not consistent with the way the GCC (and LLVM) libraries implements the functions or the way the compilers emit calls to them.

If we're going to paste in the misleading part, we should make it very clear that it is incorrect.

```suggestion
 * At the time of writing this comment, the GCC docs
 * (https://gcc.gnu.org/onlinedocs/gccint/Integer-library-routines.html)
 * show INCORRECT signatures for these library functions:
 *   int __clzsi2 (unsigned int a)
 *   int __clzdi2 (unsigned long a)
 *   int __ctzsi2 (unsigned int a)
 *   int __ctzdi2 (unsigned long a)
 * These signatures are NOT consistent with the way GCC and LLVM
 * libraries implement these functions, nor the way the compilers emit
 * calls to them.
 * In the actual libraries and compilers, the sizes of the arguments to these
 * library functions are independent of the implementation choices for
 * the sizes of `unsigned int` and `unsigned long`.
 * - `si` always signifies a 32-bit argument.
 * - `di` always signifies a 64-bit argument.
```